pandas/tests/indexes/multi/test_conversion.py

```diff
@@ -16,6 +18,40 @@ def test_to_numpy(idx):
     tm.assert_numpy_array_equal(result, exp)
 
 
+def test_array_interface(idx):
+    # https://github.com/pandas-dev/pandas/pull/60046
+    result = np.asarray(idx)
+    expected = np.empty((6,), dtype=object)
+    expected[:] = [
+        ("foo", "one"),
+        ("foo", "two"),
+        ("bar", "one"),
+        ("baz", "two"),
+        ("qux", "one"),
+        ("qux", "two"),
+    ]
+    tm.assert_numpy_array_equal(result, expected)
+
+    # it always gives a copy by default, but the values are cached, so results
+    # are still sharing memory
+    result_copy1 = np.asarray(idx)
+    result_copy2 = np.asarray(idx)
+    assert np.may_share_memory(result_copy1, result_copy2)
```

Comment:
This might be an unfortunate side effect of our caching, because a user might expect here always to get a new object. But at least when explicitly passing `copy=True`, then it appears to do the right thing. But is this numpy that does that? (still copy even if `copy=True` was passed down to `__array__`?) We should probably not keep relying on that, and do an explicit copy on our side in case of `copy=True`?

Reply:
It seems I am testing locally with numpy 2.0.2 (so not 1.x, but also not latest 2.1.x or main)

Reply:
Ah, maybe there were fixes with 2.1.